packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -765,14 +765,22 @@ class PageTransitionsTheme with Diagnosticable {
     Animation<double> secondaryAnimation,
     Widget child,
   ) {
-    TargetPlatform platform = Theme.of(context).platform;
+    final TargetPlatform platform = Theme.of(context).platform;
+
+    bool cupertinoTransitionInProgress = false;
 
     if (CupertinoRouteTransitionMixin.isPopGestureInProgress(route)) {
-      platform = TargetPlatform.iOS;
+      cupertinoTransitionInProgress = true;
+    }
+
+    PageTransitionsBuilder getTransitionBuilder() {
+      if (cupertinoTransitionInProgress) {
+        return const CupertinoPageTransitionsBuilder();
```

Comment:
Why are we forcing it to CupertinoPageTransitionsBuilder instead of looking up the iOS one in `builders` and only falling back to CupertinoPageTransitionsBuilder if there is none?

Reply:
I'm making the assumption that if a CupertinoPageTransition is in progress, than that's the builder we want to use. If it's not in progress, then we go to the builder, then the CupertinoPageTransitionsBuilder, depending on platform.

Reply:
I am not (yet) convinced that `CupertinoRouteTransitionMixin.isPopGestureInProgress(route)` returning true actually means there is a CupertinoPageTransition in progress. `CupertinoRouteTransitionMixin.isPopGestureInProgress` is implemented as `route.navigator.userGestureInProgress` - couldn't this also be set by any other custom transition? If one of those is used, we wouldn't want to change to `CupertinoPageTransitionsBuilder` mid-transition?

Reply:
I see what you are saying, but `CupertinoRouteTransitionMixin` is only added by `CupertinoPageTransitionsBuilder`. Previously it changed the platform to iOS, so theoretically we would have been getting other issues if it was changing the platform to iOS for anybody using neither `CupertinoPageTransitionsBuilder` nor `ZoomPageTransitionsBuilder`.

All that said, I only put in this logic with the boolean to cover whatever edge case setting the platform to iOS was covering before. However I have not been able to track down why it was there, and a previous PR that took this out entirely passed Google testing. I'm fine with just removing it and going straight to checking the `builders`.

Reply:
If nothing is breaking I would be in favor of taking this out.

Reply:
I was wrong, it does fail a test that changes the OS mid back gesture.